www/source/partials/docs/_using-builder-origin-keys.html.md.erb

```diff
@@ -0,0 +1,60 @@
+## <a name="origin-keys" id="origin keys" data-magellan-target="origin-keys">Origin Keys</a>
+## About origin keys
+
+Creating an origin automatically generates an origin key pair. 
+
+Origin key cryptography is asymmetric: it has a public origin key that you can distribute freely, and a private origin key that you should keep safe.
+
+Habitat Builder and Habitat Depot require access to at least the public origin key to validate packages in the origin. 
+
+Habitat Supervisors use the public origin key to verify the integrity of a downloaded Habitat artifact before running it.
+
+By default, Supervisors only run packages for which they have the public origin key, and cannot run packages without the correct origin public origin key.
+You can provide the origin public origin key to the Supervisors by pointing them to a depot that containing it, or by putting the key on disk outside of Habitat.
+
+## Creating origin keys
+
+Creating an origin, either on the Habitat Builder site or with the Habitat CLI, automatically generates two keys, a private origin key and a public origin key. You will use the private origin key to sign your packages and Supervisors will use the public origin key to verify the integrity of your packages (`.hart` files). 
+
+### Origin keys on Habitat Builder
+
+You can find your origin keys on the Builder site by navigating to the relevant origin and selecting the **Keys** tab.
+
+<img src="/images/screenshots/origin-keys.png">
+
+### Origin keys on your workstation or in the Habitat Studio
+
+You'll find any private and public origin keys that you have already saved at `~/.hab/cache/keys` on your host(local) machine, and  at `/hab/cache/keys` inside the studio environment. To see your existing origin keys, from your host command line, run:
+
+```shell
+ls -la `~~/.hab/cache/keys`
+````
+
+#### Create an origin key pair from the command line use 
+
+You have two options for creating an origin key pair from the command line. 
+
+You can create an origin key pair while setting up your host machine, following the instructions in the `hab cli setup` interaction.
+Alternatively, you can run `hab origin key generate <ORIGIN>` from either the host machine or from within the studio.
+
+## Download origin keys from Builder
+
+To download your private or public origin key, select the download icon from the right end of the key details, under the "Actions" heading.
+
+<img src="/images/screenshots/origin-keys-download.png">
+
+To get your keys from Builder from the command line, use the
+
+```shell
+hab origin key import
```

Comment:
It would be easier to use the `hab origin key download` command. There's no need to manually import keys that exist on builder.

Reply:
Done.